packages/server/src/service/util.ts

```diff
@@ -8,6 +9,45 @@ interface Options{
 }
 type AsyncFunc<A extends unknown[], R> = (...args: A) => Promise<R>
 
+/**
+ * Asynchronously reads the entire contents of a file.
+ *
+ * @param path A path to a file. If a URL is provided, it must use the
+ * file: protocol. If a file descriptor is provided, the underlying file
+ * will not be closed automatically.
+ *
+ * @param options An object that may contain an optional flag. If a flag
+ * is not provided, it defaults to 'r'.
+ */
+export const safeReadFile = async (
+  path: fs.PathLike | number,
+  options?: { encoding?: null | undefined, flag?: string | undefined },
+) => {
+  const asyncReadFile = util.promisify(fs.readFile)
```

Comment:
I don't know if this works (I just don't know enough about JS).  The old `readBinaryFile` did work because the creation of the buffer occurred during the callback, **not** after the `await` (as in this case).  Perhaps write it without `util.promisify` and use a callback so you can guarantee that the buffer is created in an atomic operation?